### ISSUE TITLE
Accesibility: Block label and title don't consider variations

### ISSUE THREAD

**Issue by t-hamano** [WordPress/gutenberg#44249]
### Description

Related to #42416

When a block variation is applied, `data-title` and `aria-label` attributes of the inserted block display information about the original block.

I think that these information should be taken into account for block variations as well as other text.

### Step-by-step reproduction instructions

- Insert `core/post-terms` (categories) block (The Category Block is a variation of the Post Terms block.).
- Check the  `data-title` and `aria-label` attributes from the browser console.

### Screenshots, screen recording, code snippet

![variation](https://user-images.githubusercontent.com/54422211/190946789-582f5897-f073-470d-864a-a8acd8a40b80.png)

### Environment info

- WordPress Version: 6.1-alpha-54198
- Gutenberg Version: 14.1.0

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes
